src/Route.js

```diff
@@ -4,60 +4,65 @@ import EventQueue from "./EventQueue";
 
 const monkeyPatchSymbol = Symbol();
 
-function Route({ path, children }) {
-  const ref = useRef({});
-  const [displayed, setDisplayed] = useState(false);
-  const [queryParams, setQueryParams] = useState({});
-  const [pathParams, setPathParams] = useState({});
-
-  useEffect(() => {
-    ref.current.path = path;
-    ref.current.displayed = displayed;
-  }, [path, displayed]);
-
-  useEffect(() => {
-    if (!window[monkeyPatchSymbol]) {
-      window[monkeyPatchSymbol] = new EventQueue();
-
-      const doProxify = proxify(() => window[monkeyPatchSymbol].broadcast());
-
-      doProxify("pushState");
-      doProxify("back");
-      doProxify("forward");
-      doProxify("go");
-      doProxify("replaceState");
+function routeFactory(notFound) {
+  return function Route({ path, children }) {
+    const ref = useRef({});
+    const [displayed, setDisplayed] = useState(false);
+    const [queryParams, setQueryParams] = useState({});
+    const [pathParams, setPathParams] = useState({});
+
+    useEffect(() => {
+      ref.current.path = path;
+      ref.current.displayed = displayed;
+    }, [path, displayed]);
+
+    useEffect(() => {
+      setupIfNeeded();
+
+      const updateCallback = anotherRouteMatched => {
+        let matchedRoute = false;
```

Comment:
Maybe you can add a little "optimisation" here?
`if (notFound && anotherRouteMatched === undefined) return;`